### PR TITLE
Update the ppp documentation

### LINKED ISSUE #67171: net: ppp: doc: PPP documentation page ('connectivity/networking/api/ppp') needs updating
**Describe the bug**

Spotted two issues with [PPP documentation page](https://docs.zephyrproject.org/latest/connectivity/networking/api/ppp.html) while looking for dead links in the documentation:

- "See also the [samples/net/sockets/echo_server/overlay-ppp.conf](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/net/sockets/echo_server/overlay-ppp.conf) file for configuration option examples." is a dead link.
- "The PPP support in Zephyr 2.0 is still experimental [...]" sentence needs updating

**Environment**

* Page: `connectivity/networking/api/ppp`
* Version: main
* SHA-1: 69d19ae9fb68d0be38ceb22ba43c9fff985cbbaa

### PR DESCRIPTION
**Pull request by jukkar**
Fixes #67171

### CODE REVIEW

File: doc/connectivity/networking/api/gsm_modem.rst
Comment (kartben) [inline doc/connectivity/networking/api/gsm_modem.rst:1]: please introduce an appropriate redirect for this page for folks browsing old versions of the documentation (in `doc/_scripts/redirects.py`)

**Review by kartben — CHANGES_REQUESTED** [review]
Thanks!

"The actual gsm-modem sample was removed earlier [...]" - was it / should it be? AFAIKT it is still there?

**Comment by jukkar** [comment]
> "The actual gsm-modem sample was removed earlier [...]" - was it / should it be? AFAIKT it is still there?

Ah, yes. It was deprecated but not yet removed.

**Comment by kartben** [comment]
> > "The actual gsm-modem sample was removed earlier [...]" - was it / should it be? AFAIKT it is still there?
> 
> Ah, yes. It was deprecated but not yet removed.

so maybe it's also premature to delete `doc/connectivity/networking/api/gsm_modem.rst`? I will leave it up to decide what you (and maybe @bjarki-trackunit) think is best

**Review by kartben — APPROVED** [review]
👍🏼

**Review by rlubos — APPROVED** [review]

**Pull request merged**